src/lib.rs

```diff
@@ -776,6 +779,88 @@ impl Client {
         })
     }
 
+    /// Gets the current window handle.
+    pub async fn get_window_handle(
+        &mut self,
+    ) -> Result<webdriver::common::WebWindow, error::CmdError> {
+        let res = self.issue(WebDriverCommand::GetWindowHandle).await?;
+        match res {
+            Json::String(x) => Ok(webdriver::common::WebWindow(x)),
+            v => Err(error::CmdError::NotW3C(v)),
+        }
+    }
+
+    /// Gets a list of all active windows (and tabs)
+    pub async fn get_window_handles(
+        &mut self,
+    ) -> Result<Vec<webdriver::common::WebWindow>, error::CmdError> {
+        let res = self.issue(WebDriverCommand::GetWindowHandles).await?;
+        match res {
+            Json::Array(handles) => handles
+                .into_iter()
+                .map(|handle| match handle {
+                    Json::String(x) => Ok(webdriver::common::WebWindow(x)),
+                    v => Err(error::CmdError::NotW3C(v)),
+                })
+                .collect::<Result<Vec<_>, _>>(),
+            v => Err(error::CmdError::NotW3C(v)),
+        }
+    }
+
+    /// Switches to the chosen window.
+    pub async fn switch_to_window(
+        &mut self,
+        window: webdriver::common::WebWindow,
+    ) -> Result<(), error::CmdError> {
+        let params = SwitchToWindowParameters { handle: window.0 };
+        let _res = self.issue(WebDriverCommand::SwitchToWindow(params)).await?;
+        Ok(())
+    }
+
+    /// May close the session if no other windows exist.
+    pub async fn close_window(&mut self) -> Result<(), error::CmdError> {
+        let _res = self.issue(WebDriverCommand::CloseWindow).await?;
+        Ok(())
+    }
+
+    /// Creates a new window. If `is_tab` is `true`, then a tab will be created instead.
+    ///
+    /// Requires geckodriver > 0.24 and firefox > 66
+    ///
+    /// Windows are treated the same as tabs by the webdriver protocol.
+    /// The functions `new_window`, `switch_to_window`, `close_window`, `get_window_handle` and `get_window_handles`
+    /// all operate on both tabs and windows.
+    pub async fn new_window(
+        &mut self,
+        is_tab: bool,
```

Comment:
Would this make more sense as `as_tab`, since the window has not yet been created?